Design/Instrument-2.0/requirements-v2.md

```diff
@@ -1,15 +1,34 @@
 Suggestions and Requirements for Future Features
 ================================================
 
-### General
+##Motivation##
 
-* The design should be optimized for reads, not for writes, since the former is much more frequent.
+The current instrument geometry matched the requirements in the early programme lifecycle. We have subseqently retro-fitted the geometry as more instruments and more facilities joined the project. This has worked well, but we have several major issues to solve for which a deeper and thourough rexeamination of the existing implementation is required. Firstly, experimental setup is becoming increasingly complex, and we need a virutal instrument that can model the real experimental setup. Secondly, the performance demands are now much greater than orignally designed for, particularly with the live-reduction challenge forethcoming at the ESS. Thirdly, the current syntax for describing instruments is not as logical or straight forward as it ought to be, and encourages instrument related errors in the reduced data.
 
-### Moving Detectors
+##High-level Functional Requirements##
 
-TODO
+###Mandatory###
 
-* For this it might be useful to separate things into two different concepts: (1) positions/rotations of spectra/event-list (2) detectors with no position/rotation information.
+* Existing functionality listed [here](https://github.com/mantidproject/documents/blob/Instrument-Geometry/Design/Instrument-2.0/features-v1.md) should be preserved.
+* The design should simplify the process of directing and configuraton from the current experiment (Instrument control).
+* The design should allow for complex beam paths, where components may order themselves to create the l1 & l2.
+* The design should allow for indirect secondary (i.e. after sample) beam paths, where the detectors and components may have different positions in real and 'neutronic' space.
+* Any new schema should allow for thorough validation to detect any logical errrors, for example, avoiding component collision
+* Instrument parameter syntax should be richer, particularly when labelling and applying multiple functions to components.
+* Any new schema should be more cohesive and self-describing than the current system
+* The design should be optimized for reads, not for writes, since the former is much more frequent. See non-functional requriements.
+* The design should allows for Moving instruments. This includes instruments that move between acquisition scan points and continuous motion during acquisition. For this it might be useful to separate things into two different concepts: (1) positions/rotations of spectra/event-list (2) detectors with no position/rotation information.
```

Comment:
@stuartcampbell I have read this [document](https://github.com/mantidproject/documents/edit/Instrument-Geometry/Design/HandlingMovingInstruments.md) and come to the conclusion that your sentence here mostly captures those unique use-cases.